### PR TITLE
Update dependencies, use latest psalm, apply PSR-2

### PR DESCRIPTION
**Pull request by andrew-demb**
My fixer tried to rearrange `throws`-annotations (throws as latest in phpdoc block), but I ignore this.

Also for simplify I ignore fixer add an empty line for single-line functions in specs (example: https://github.com/prewk/result/pull/6/files#diff-15a90b5b523550fe618e12b615745b04R35).

### CODE REVIEW

**Comment by andrew-demb** [comment]
As next, we can 
- apply PSR-12 rules
- try to cover more types by psalm (psalm improved since 1.x versions for closures and templates params)

File: spec/Result/OkSpec.php [hunk 1]
```diff
@@ -42,7 +42,8 @@ function it_maps()
     function it_doesnt_errMap()
     {
         $this->beConstructedWith("foo");
-        $result = $this->mapErr(function($value) {});
+        $result = $this->mapErr(function ($value) {
+        });
```
Comment (prewk) [inline spec/Result/OkSpec.php]: Expanding empty closures from `{}` to `{\n}` is a bit ugly but not a hill I would die on ^^

File: spec/Result/OkSpec.php [hunk 2]
```diff
@@ -100,7 +104,8 @@ function it_unwrapOrs_with_its_value()
     function it_unwrapOrElses_with_its_value()
     {
         $this->beConstructedWith("value");
-        $this->unwrapOrElse(function() {})->shouldBe("value");
+        $this->unwrapOrElse(function () {
+        })->shouldBe("value");
```
Comment (prewk) [inline spec/Result/OkSpec.php]: Especially on places like this. Is it a fixer thing that causes this?

Comment (andrew-demb) [reply spec/Result/OkSpec.php]: Yes. PSR declares newlines for function body.
I think we can skip this rules for specs.

Comment (prewk) [reply spec/Result/OkSpec.php]: I see. I don't mind.

**Comment by prewk** [comment]
Thanks again, great stuff :)

**Pull request merged**